### PR TITLE
[MRG] viz.plot_bem():  include brain surfaces

### PR DESCRIPTION
**Pull request by christianbrodbeck**
I wanted to plot a brain (white matter) surface along with the other bem surfaces, to this end I modified `viz.plot_bem` by adding a `brain_surfaces` argument. If this is generally useful (and there isn't already another preferred way to do this) and you agree with the API I will document it and make this PR ready.

`mne.viz.plot_bem(subject, subjects_dir, brain_surfaces='white')`
 
00dd00
(updated)

### CODE REVIEW

**Comment by agramfort** [comment]
looks useful indeed

while you're at it can you make it consistent with the colors of the mne command. Just try with:


```mne freeview_bem_surfaces -s sample -m flash```

**Comment by dengemann** [comment]
This is really nice! +1

**Comment by Eric89GXL** [comment]
Looks great. Any way you could take care of #3278, too, somehow? That talks about `plot_trans` but `plot_bem` is a fine place, too

File: mne/viz/misc.py [hunk 1]
```diff
@@ -311,19 +312,19 @@ def _plot_mri_contours(mri_fname, surf_fnames, orientation='coronal',
...
             elif orientation == 'axial':
                 ax.tricontour(surf['rr'][:, 2], surf['rr'][:, 0],
                               surf['tris'], surf['rr'][:, 1],
-                              levels=[sl], colors='yellow', linewidths=2.0)
+                              levels=[sl], colors=color, linewidths=2.0)
             elif orientation == 'sagittal':
                 ax.tricontour(surf['rr'][:, 2], surf['rr'][:, 1],
                               surf['tris'], surf['rr'][:, 0],
-                              levels=[sl], colors='yellow', linewidths=2.0)
+                              levels=[sl], colors=color, linewidths=2.0)
```
Comment (jona-sassenhagen) [inline mne/viz/misc.py]: Couldn't the coronal/axial/sagittal branching be reduced to make this a bit more concise? E.g.,

```
one = 0 if orientation == 'coronal' else 1
two = dict(coronal=2, axial=1, sagittal=2)[orientation]
ax.tricontour(surf['rr'][:, 2], surf['rr'][:, one],
     surf['tris'], surf['rr'][:, two],
     levels=[sl], colors=color, linewidths=2.0)
```

File: mne/viz/misc.py [hunk 2]
```diff
@@ -367,20 +368,32 @@ def plot_bem(subject=None, subjects_dir=None, orientation='coronal',
     if not op.isdir(bem_path):
         raise IOError('Subject bem directory "%s" does not exist' % bem_path)
 
-    surf_fnames = []
+    surfaces = {}
```
Comment (jona-sassenhagen) [inline mne/viz/misc.py]: I think we always do `dict()`

File: mne/viz/misc.py [hunk 3]
```diff
@@ -367,20 +368,32 @@ def plot_bem(subject=None, subjects_dir=None, orientation='coronal',
...
+            surfaces[surf_fname] = 'yellow'
+
+    if brain_surfaces is not None:
+        if isinstance(brain_surfaces, string_types):
+            brain_surfaces = (brain_surfaces,)
+        for surf_name in brain_surfaces:
+            for hemi in ('lh', 'rh'):
+                surf_fname = op.join(subjects_dir, subject, 'surf',
+                                     hemi + '.' + surf_name)
+                if op.exists(surf_fname):
```
Comment (jona-sassenhagen) [inline mne/viz/misc.py]: My intuition would be to use try/except here ... doesn't require evaluating this expression, and is less words.

**Comment by christianbrodbeck** [comment]
@Eric89GXL do you have a specific visualization in mind? All the points that fall into a slice? If displaying a source space it might also be desirable to plot normals?

**Comment by christianbrodbeck** [comment]
@agramfort colors are updated in the first post. Do you have specific colors in mind for the brain surfaces? We could also expose the colors in `surfaces` and `brain_surfaces` parameters that can be specified as `{surface_name: color}` dictionaries.

File: mne/viz/misc.py (hunk 3 above)
Comment (christianbrodbeck) [inline mne/viz/misc.py]: What would you try? We're not reading any files yet.

**Comment by Eric89GXL** [comment]
Yeah just the locations, at this resolution normals is probably too
ambitious. Just want to give people a sense (e.g. to check alignment) more
than give all information. plot_trans eventually could do locations and
normals since it uses mayavi.

**Comment by agramfort** [comment]
don't expose colors. I feel it's an overkill.

I would put the cortex in light brigth green as in the freesurfer convention.

travis is not happy

**Comment by christianbrodbeck** [comment]
@agramfort I did not see the green in freeview, would it be `#00FF00`?

**Comment by agramfort** [comment]
I mean in the logo:

http://miykael.github.io/nipype-beginner-s-guide/_images/logoFreeSurfer.jpg

File: mne/viz/misc.py (hunk 3 above)
Comment (jona-sassenhagen) [inline mne/viz/misc.py]: Ah right

**Comment by christianbrodbeck** [comment]
Very green or slightly less green? :)

00ff00
00dd00

**Comment by agramfort** [comment]
The lesser Green. Btw you have an intersection pb with red and yellow. Not good for EEG 






On Tue, Jul 19, 2016 at 3:56 PM +0200, "Christian Brodbeck" <notifications@github.com> wrote:












Very green or slightly less green? :)

**Comment by agramfort** [comment]
MRG or WIP?

**Comment by christianbrodbeck** [comment]
The surface plotting is ready, I was going to see if I can add source spaces as sugested by @Eric89GXL but I could also open a new PR.

> Btw you have an intersection pb with red and yellow. Not good for EEG 

Thanks, only doing MEG :) , although the reason for looking into this was another subject with a mistake in the inner_skull, but I haven't had time to investigate how to fix it yet

**Comment by agramfort** [comment]
please add a test and set PR to MRG

**Comment by christianbrodbeck** [comment]
> please add a test and set PR to MRG

Ok, this is ready, I will update #3450 after this is merged.

File: mne/viz/misc.py [hunk 4]
```diff
@@ -237,17 +238,17 @@ def plot_source_spectrogram(stcs, freq_bins, tmin=None, tmax=None,
...
     """Plot BEM contours on anatomical slices.
 
     Parameters
     ----------
     mri_fname : str
         The name of the file containing anatomical data.
-    surf_fnames : list of str
-        The filenames for the BEM surfaces in the format
-        ['inner_skull.surf', 'outer_skull.surf', 'outer_skin.surf'].
+    surfaces : dict
```
Comment (Eric89GXL) [inline mne/viz/misc.py]: dict iteration order is not guaranteed, maybe list of tuple is better?

File: mne/viz/misc.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/viz/misc.py]: this comment has not been addressed

File: mne/viz/misc.py [hunk 5]
```diff
@@ -346,6 +347,10 @@ def plot_bem(subject=None, subjects_dir=None, orientation='coronal',
         'coronal' or 'axial' or 'sagittal'.
     slices : list of int
         Slice indices.
+    brain_surfaces : None | str | list of str
+        One or more brain surface to plot (optional). Entries should correspond
+        to files in the subject's ``surf`` directory, e.g., ``"white"`` will
+        plot ``<subject>/surf/lh.white`` and ``<subject>/surf/rh.white``.
```
Comment (Eric89GXL) [inline mne/viz/misc.py]: I don't think you need to give paths, we talk about these surfaces in a lot of places

**Comment by Eric89GXL** [comment]
Enhancements to `plot_bem` should probably also be passed on to `Report` e.g. as options, since I think people use it there frequently

Otherwise LGTM

File: mne/viz/misc.py (hunk 1 above)
Comment (christianbrodbeck) [inline mne/viz/misc.py]: I find the flat code pretty readable, probably more readable than if I would have to look up indexing operations in a different place, but if a majority prefer I can refactor it

**Comment by christianbrodbeck** [comment]
> Enhancements to plot_bem should probably also be passed on to Report e.g. as options, since I think people use it there frequently

I looked into this quickly and it does not look straight forward because `Report` does not seem to share code with `plot_bem`, or did I miss something?

**Comment by agramfort** [comment]
CIs are not happy

**Comment by christianbrodbeck** [comment]
> CIs are not happy

They all crashed with messages like `The command "conda install --yes --quiet $ENSURE_PACKAGES nose coverage" failed and exited with 1 during .`, is there a way to restart them without adding a new commit?

**Comment by Eric89GXL** [comment]
It has been like that for the last few hours, I've tried restarting, doesn't help. We'll have to wait :(

**Comment by Eric89GXL** [comment]
Travis is back up, I restarted the builds

**Comment by christianbrodbeck** [comment]
AppVeyor still has a strange setup error

**Comment by Eric89GXL** [comment]
I couldn't restart that one but I think it should be fine if Travis is happy

**Comment by agramfort** [comment]
just restarted appveyor

**Comment by agramfort** [comment]
thx @christianbrodbeck

**Pull request merged**